exasol/analytics/udf/communication/peer_communicator/background_listener_thread.py

```diff
@@ -411,18 +434,24 @@ def _create_register_peer_connection(self, message: messages.RegisterPeer):
             )
         else:
             predecessor_send_socket_factory = None
-        self._register_peer_connection = RegisterPeerConnection(
+        return RegisterPeerConnection(
             predecessor=message.source,
             predecessor_send_socket_factory=predecessor_send_socket_factory,
             successor=message.peer,
             successor_send_socket_factory=successor_send_socket_factory,
             my_connection_info=self._my_connection_info,
         )
 
+    @property
```

Comment:
Property method `register_peer_connection` checks the private attribute `_register_peer_connection` and raises an exception when it is `None` to avoid code duplication.

Questions: 
* Q1) The names of the private attribute and the property differ on in the prefix `_` and potentially could be mixed up. Should we use different names that can easier be distinguished? 
* Q2) The property is public. Should it be private?

Reply:
See last push